addons/calendar/models/calendar_event.py

```diff
@@ -741,7 +741,7 @@ def create(self, vals_list):
                 if values.get('activity_ids'):
                     continue
                 res_model_id = values.get('res_model_id', defaults.get('res_model_id'))
-                res_id = values.get('res_id', defaults.get('res_id'))
+                values['res_id'] = res_id = values.get('res_id') if values.get('res_id') else defaults.get('res_id')
```

Comment:
```suggestion
                values['res_id'] = res_id = values.get('res_id') or defaults.get('res_id')
```

Could you please add a python test ? So that the fix is not broken during forward port.

Thanks.